### PR TITLE
#1989 hard deleted delete manufacturer endpoint

### LINKED ISSUE #1989: Delete Manufacturer Endpoint Should be Hard Delete
### Description

Right now we soft delete manufacturers in the database (mark it as deleted by setting a dateDeleted on it), however, because we use name as the id, this would make it so that you could never make a manufacturer with that name again once it's deleted. We should instead just have people hard delete the manufacturer from the database and make a new one.

### Acceptance Criteria

- [x] The manufacturer service method should hard delete manufacturers rather than setting a dateDeleted on it.
- [x] Update any related tests

### PR DESCRIPTION
**Pull request by AShyamkumar19**
## Changes

Made the delete manufacturer a hard delete 

## Test Cases
- Made a postman test to check if it hard deleted
- made a test to make sure it throws an exception when a manufacturer is trying to get deleted while it still has materials

## Description
- Added seed data in projects.test-data
     - created local Finishline data on manufacturer and material and changed the original manufacturer to prismaManufacturer
- Deleted dateDeleted attribute of the manufacturer in the schema and created a migration
     - We deleted it because we are hard deleting dataDeleted from the manufacturer so we don't need that variable anymore
- Tested on Postman to see if a manufacturer is hard deleted:

Screenshot 2024-02-06 205701

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1989

### CODE REVIEW

File: src/backend/src/services/projects.services.ts [hunk 1]
```diff
@@ -751,13 +751,10 @@ export default class ProjectsService {
 
     if (manufacturer.dateDeleted) throw new DeletedException('Manufacturer', manufacturer.name);
 
-    const dateDeleted: Date = new Date();
-    const deletedManufacturer = await prisma.manufacturer.update({
+    //const dateDeleted: Date = new Date();
```
Comment (Aaryan1203) [inline src/backend/src/services/projects.services.ts:754]: delete this

File: src/backend/src/services/projects.services.ts [hunk 2]
```diff
@@ -751,13 +751,10 @@ export default class ProjectsService {
 
     if (manufacturer.dateDeleted) throw new DeletedException('Manufacturer', manufacturer.name);
```
Comment (Aaryan1203) [inline src/backend/src/services/projects.services.ts:752]: no need to check for this anymore because the manufacturers are no longer being soft deleted

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
left some comments but good work!

**Review by Aaryan1203 — COMMENTED** [review]
LGTM! Great work! 

@RChandler234 should we be testing all endpoints manually using postman so this type of problem doesn't occur again where an endpoint doesn't work as expected cuz sometimes the tests don't really help?

**Comment by RChandler234** [comment]
> LGTM! Great work!
> 
> @RChandler234 should we be testing all endpoints manually using postman so this type of problem doesn't occur again where an endpoint doesn't work as expected cuz sometimes the tests don't really help

yes 100%, this one was likely my bad cause some of the BOM stuff was done at the last minute, in general we should be thoroughly manually testing everything

File: src/backend/src/services/projects.services.ts [hunk 3]
```diff
@@ -749,15 +749,9 @@ export default class ProjectsService {
...
-    const dateDeleted: Date = new Date();
-    const deletedManufacturer = await prisma.manufacturer.update({
+    const deletedManufacturer = await prisma.manufacturer.delete({
       where: {
         name: manufacturer.name
-      },
-      data: {
-        dateDeleted
       }
     });
```
Comment (RChandler234) [inline src/backend/src/services/projects.services.ts:756]: can you wrap the deletedManufacturer being returned in the manufacturerTransformer

**Review by RChandler234 — CHANGES_REQUESTED** [review]
while you're in here making the change, try manually testing the endpoint, this video will show you how: https://www.youtube.com/watch?v=CgFTb_hKdEo&feature=youtu.be

**Review by RChandler234 — CHANGES_REQUESTED** [review]
also throw an error if the manufacturer is associated with any materials, because otherwise prisma will throw an error when you try to delete it

**Review by Aaryan1203 — APPROVED** [review]
LGTM!

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**